flexget/plugins/filter/unique.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import unicode_literals, division, absolute_import
+import logging
+
+from flexget import plugin
+from flexget.event import event
+
+log = logging.getLogger('unique')
+
+
+class Unique(object):
+    """
+    Take action on entries with duplicate fields, except for the first item
+
+    Reject the second+ instance of every movie:
+
+      unique:
+        field:
+          - imdb_id
+          - movie_name
+        action: reject
+    """
+
+    schema = {
+        'oneOf': [
+            {'type': 'boolean'},
+            {
+                'type': 'object',
+                'properties': {
+                    'field': {
+                        'oneOf': [
+                            {'type': 'string'},
+                            {'type': 'array'},
+                        ]
+                    },
+                    'action': {'enum': ['accept', 'reject']},
+                },
+                'required': [],
+                'additionalProperties': False
+            },
+        ],
+    }
+
+    def prepare_config(self, config):
+        if isinstance(config, bool) or config is None:
+            config = {}
+        if 'action' not in config:
+            config['action'] = 'reject'
+        if 'field' not in config:
+            config['field']=[]
+        return config
+
+    def extract_fields(self, entry, field_names):
+        if field_names:
+            pass
+            return {field: entry.get(field) for field in field_names}
+        else:
+            return entry
+
+    def should_ignore(self, item, action):
+        return item.accepted and action == 'accept' or\
+               item.rejected and action == 'reject'
+
+    def on_task_filter(self, task, config):
+        config = self.prepare_config(config)
+        field_names = config['field']
+        # Convert to list
+        if not isinstance(field_names, list):
+            field_names = [field_names]
+        action = config['action']
+        entries = list(task.entries)
+        for i, entry in enumerate(entries):
+            entry_fields = self.extract_fields(entry,field_names)
+            # Ignore already processed entries
+            if self.should_ignore(entry, action): continue
+            # Iterate over next items, try to find a similar item
+            for prospect in entries[i+1:]:
+                prospect_fields = self.extract_fields(prospect,field_names)
+                # Ignore processed prospects
+                if self.should_ignore(prospect, action): continue
+                if entry_fields == prospect_fields and any(entry_fields):
```

Comment:
Isn't `any(entry_fields)` the same as `entry_fields`? You can also short-circuit this if by checking entry_fields first ie. `if entry_fields and entry_fields == prospect_fields`.